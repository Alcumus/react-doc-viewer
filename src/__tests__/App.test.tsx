import { act, render } from "@testing-library/react";
import React from "react";
import DocViewer from "..";

test("renders component with no documents", () => {
  const comp = render(<DocViewer documents={[]} />);
  expect(comp?.getByTestId("react-doc-viewer")).toMatchSnapshot();
});

test("renders component with no file name", async () => {
  let comp = render(
    <DocViewer
      documents={[{ uri: require("../_example-files_/pdf.pdf") }]}
      config={{ header: { disableFileName: true } }}
    />
  );

  act(async () => {
    await comp?.findByTestId("file-name");
  });

  expect(comp?.getByTestId("react-doc-viewer")).toMatchSnapshot();
});

test("renders component with no header", async () => {
  let comp = render(
    <DocViewer
      documents={[{ uri: require("../_example-files_/pdf.pdf") }]}
      config={{ header: { disableHeader: true } }}
    />
  );

  act(async () => {
    await comp?.findByTestId("header-bar");
  });

  expect(comp?.getByTestId("react-doc-viewer")).toMatchSnapshot();
});
