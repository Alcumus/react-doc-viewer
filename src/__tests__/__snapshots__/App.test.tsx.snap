// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component with no documents 1`] = `
<div
  class="sc-fznWqX cxRsBy"
  data-testid="react-doc-viewer"
  id="react-doc-viewer"
>
  <div
    class="sc-Axmtr geoSRH"
    data-testid="header-bar"
    id="header-bar"
  />
  <div
    class="sc-AxmLO bNNfcA"
    id="proxy-renderer"
  >
    <div
      id="no-documents"
    />
  </div>
</div>
`;

exports[`renders component with no file name 1`] = `
<div
  class="sc-fznWqX cxRsBy"
  data-testid="react-doc-viewer"
  id="react-doc-viewer"
>
  <div
    class="sc-Axmtr geoSRH"
    data-testid="header-bar"
    id="header-bar"
  />
  <div
    class="sc-AxmLO bNNfcA"
    id="proxy-renderer"
  >
    <div
      class="sc-fzozJi hXCnGH"
      data-testid="loading-renderer"
      id="loading-renderer"
    >
      <div
        class="sc-fzoLsD dachMJ"
      >
        <svg
          height="40"
          id="Icons"
          style="align-self: center; justify-self: center;"
          version="1.1"
          viewBox="0 0 32 32"
          width="40"
          xml:space="preserve"
        >
          <g>
            <path
              d="M16,2c-0.6,0-1,0.4-1,1v5c0,0.6,0.4,1,1,1s1-0.4,1-1V3C17,2.4,16.6,2,16,2z"
              fill="#444"
            />
            <path
              d="M7.5,6.1c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l3.5,3.5c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3c0.4-0.4,0.4-1,0-1.4L7.5,6.1 z"
              fill="#444"
            />
            <path
              d="M9,16c0-0.6-0.4-1-1-1H3c-0.6,0-1,0.4-1,1s0.4,1,1,1h5C8.6,17,9,16.6,9,16z"
              fill="#444"
            />
            <path
              d="M9.6,20.9l-3.5,3.5c-0.4,0.4-0.4,1,0,1.4c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3l3.5-3.5c0.4-0.4,0.4-1,0-1.4 S10,20.6,9.6,20.9z"
              fill="#444"
            />
            <path
              d="M16,23c-0.6,0-1,0.4-1,1v5c0,0.6,0.4,1,1,1s1-0.4,1-1v-5C17,23.4,16.6,23,16,23z"
              fill="#444"
            />
            <path
              d="M22.4,20.9c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l3.5,3.5c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3c0.4-0.4,0.4-1,0-1.4 L22.4,20.9z"
              fill="#444"
            />
            <path
              d="M29,15h-5c-0.6,0-1,0.4-1,1s0.4,1,1,1h5c0.6,0,1-0.4,1-1S29.6,15,29,15z"
              fill="#444"
            />
            <path
              d="M21.7,11.3c0.3,0,0.5-0.1,0.7-0.3l3.5-3.5c0.4-0.4,0.4-1,0-1.4s-1-0.4-1.4,0l-3.5,3.5c-0.4,0.4-0.4,1,0,1.4 C21.1,11.2,21.4,11.3,21.7,11.3z"
              fill="#444"
            />
          </g>
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders component with no header 1`] = `
<div
  class="sc-fznWqX cxRsBy"
  data-testid="react-doc-viewer"
  id="react-doc-viewer"
>
  <div
    class="sc-AxmLO bNNfcA"
    id="proxy-renderer"
  >
    <div
      class="sc-fzozJi hXCnGH"
      data-testid="loading-renderer"
      id="loading-renderer"
    >
      <div
        class="sc-fzoLsD dachMJ"
      >
        <svg
          height="40"
          id="Icons"
          style="align-self: center; justify-self: center;"
          version="1.1"
          viewBox="0 0 32 32"
          width="40"
          xml:space="preserve"
        >
          <g>
            <path
              d="M16,2c-0.6,0-1,0.4-1,1v5c0,0.6,0.4,1,1,1s1-0.4,1-1V3C17,2.4,16.6,2,16,2z"
              fill="#444"
            />
            <path
              d="M7.5,6.1c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l3.5,3.5c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3c0.4-0.4,0.4-1,0-1.4L7.5,6.1 z"
              fill="#444"
            />
            <path
              d="M9,16c0-0.6-0.4-1-1-1H3c-0.6,0-1,0.4-1,1s0.4,1,1,1h5C8.6,17,9,16.6,9,16z"
              fill="#444"
            />
            <path
              d="M9.6,20.9l-3.5,3.5c-0.4,0.4-0.4,1,0,1.4c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3l3.5-3.5c0.4-0.4,0.4-1,0-1.4 S10,20.6,9.6,20.9z"
              fill="#444"
            />
            <path
              d="M16,23c-0.6,0-1,0.4-1,1v5c0,0.6,0.4,1,1,1s1-0.4,1-1v-5C17,23.4,16.6,23,16,23z"
              fill="#444"
            />
            <path
              d="M22.4,20.9c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l3.5,3.5c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3c0.4-0.4,0.4-1,0-1.4 L22.4,20.9z"
              fill="#444"
            />
            <path
              d="M29,15h-5c-0.6,0-1,0.4-1,1s0.4,1,1,1h5c0.6,0,1-0.4,1-1S29.6,15,29,15z"
              fill="#444"
            />
            <path
              d="M21.7,11.3c0.3,0,0.5-0.1,0.7-0.3l3.5-3.5c0.4-0.4,0.4-1,0-1.4s-1-0.4-1.4,0l-3.5,3.5c-0.4,0.4-0.4,1,0,1.4 C21.1,11.2,21.4,11.3,21.7,11.3z"
              fill="#444"
            />
          </g>
        </svg>
      </div>
    </div>
  </div>
</div>
`;
